frontend/src/components/Contexts/RobotContext.tsx

```diff
@@ -60,6 +66,23 @@ export const RobotProvider: FC<Props> = ({ children }) => {
                     return [...oldRobotListCopy]
                 })
             })
+            registerEvent(SignalREventLabels.robotPropertyUpdated, (username: string, message: string) => {
+                const robotPropertyUpdate: RobotPropertyUpdate = JSON.parse(message)
+                setEnabledRobots((oldRobotList) => {
+                    const oldRobotListCopy = [...oldRobotList]
+                    const index = oldRobotListCopy.findIndex((r) => r.id === robotPropertyUpdate.robotId)
+                    if (index > -1) {
+                        const robot = oldRobotListCopy[index]
+                        if (robotPropertyUpdate.propertyName in robot) {
+                            ;(robot as any)[robotPropertyUpdate.propertyName] = robotPropertyUpdate.propertyValue
```

Comment:
I'm not familiar with this syntax but I trust you know what you're doing

Reply:
Couldn't get typescript to accept that propertyName is a valid property name so had to use any type to get around it